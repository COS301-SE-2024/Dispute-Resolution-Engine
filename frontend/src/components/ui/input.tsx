import * as React from "react";

import { cn } from "@/lib/utils";

export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <input
        type={type}
        className={cn(
          "flex w-full rounded-md border py-4 px-3 border-zinc-200 bg-white  text-sm ring-offset-white file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-zinc-500 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-zinc-950 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 dark:border-dre-100/20 dark:bg-dre-bg-dark dark:ring-offset-zinc-950 dark:placeholder:text-white/50 dark:focus-visible:ring-zinc-300",
          className
        )}
        ref={ref}
        {...props}
      />
    );
  }
);
Input.displayName = "Input";

export { Input };
