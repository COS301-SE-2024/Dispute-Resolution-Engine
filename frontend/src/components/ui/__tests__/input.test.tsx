import renderer from 'react-test-renderer';
import { Input } from '../input';

it('renders correctly', () => {
  const tree = renderer
    .create(<Input />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
