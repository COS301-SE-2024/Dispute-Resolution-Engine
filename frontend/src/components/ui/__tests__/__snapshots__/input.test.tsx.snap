// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="flex w-full rounded-md border py-4 px-3 border-zinc-200 bg-white text-sm ring-offset-white file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-zinc-500 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-zinc-950 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 dark:border-dre-100/20 dark:bg-dre-bg-dark dark:ring-offset-zinc-950 dark:placeholder:text-white/50 dark:focus-visible:ring-zinc-300"
/>
`;
